tests/test_register.py

```diff
@@ -405,14 +405,17 @@ def assert_ineq(left, right):
 def test_equality_function():
     reg1 = Register({"c": (1, 2), "d": (8, 4)})
     assert_eq(reg1, reg1)
-    assert_eq(reg1, Register({"d": (8, 4), "c": (1, 2)}))
```

Comment:
Shouldn't this still be true?

Reply:
These register's ids don't have the same order, so they should not be equal.